pkg/cmd/server/api/helpers.go

```diff
@@ -17,6 +17,69 @@ import (
 	cmdutil "github.com/openshift/origin/pkg/cmd/util"
 )
 
+var (
+	knownOpenShiftFeatureSet map[string]string
+)
+
+func init() {
+	knownOpenShiftFeatureSet = make(map[string]string, len(KnownOpenShiftFeatures))
+	for _, feature := range KnownOpenShiftFeatures {
+		knownOpenShiftFeatureSet[strings.ToLower(feature)] = feature
+	}
+}
+
+// Add extends feature list with given valid items. They are appended
+// unless already present.
+func (fl *FeatureList) Add(items ...string) error {
+	unknown := []string{}
+	for _, item := range items {
+		if fl.Has(item) {
+			continue
+		}
+		feature, exists := knownOpenShiftFeatureSet[strings.ToLower(item)]
```

Comment:
I thought it wasn't an issue any more because `fl` wouldn't have been updated if any *uknown` had been found. But prepending it before `Has` check may be more effective - done.